demos/using-dev-build/test.ts

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env ts-node
+/* This starts the proxy and the test suite. The proxy configuration is read
+ * from dfx.json and canister_ids.json. The proxy is shutdown after the tests
+ * are run.
+ * This expects the replica to be running, and expects the II canister to have
+ * been deployed.
+ */
+
+import * as fs from "fs";
+import { ChildProcess, spawn } from "child_process";
+
+// Port and URL for II and the sign in app, then exported to the tests
+const II_DAPP_PORT = 8086;
+const II_DAPP_URL = `http://localhost:${II_DAPP_PORT}`;
+
+const WEBAPP_PORT = 8087;
+const WEBAPP_URL = `http://localhost:${WEBAPP_PORT}`;
+
+type Arguments = { noRun: boolean };
+
+/** Parse CLI arguments
+ */
+function parseArguments(): Arguments {
+  if (!(process.argv.length == 2 || process.argv.length == 3)) {
+    throw new Error("Usage: test.ts [--no-run]");
+  }
+
+  if (process.argv.length == 3) {
+    if (process.argv[2] !== "--no-run") {
+      throw new Error("Usage: test.ts [--no-run]");
+    }
+
+    return { noRun: true };
+  } else {
+    return { noRun: false };
+  }
+}
+
+type CanisterIDs = { webapp: string; internetIdentity: string };
+
+/*
+ * Read the values from dfx.json and canister_ids.json
+ */
+function parseCanisterIDs(): CanisterIDs {
+  const CANISTER_IDS_PATH = `${__dirname}/.dfx/local/canister_ids.json`;
+  try {
+    const canister_ids = JSON.parse(fs.readFileSync(CANISTER_IDS_PATH, "utf8"));
+    return {
+      internetIdentity: canister_ids["internet_identity"].local,
+      webapp: canister_ids["webapp"].local,
+    };
+  } catch (e) {
+    console.log(
+      `Could not read 'internet_identity' and 'webapp' local canister IDs from ${CANISTER_IDS_PATH}`
+    );
+    throw e;
+  }
+}
+
+function parseReplicaHost(): string {
```

Comment:
it actually returns `string | undefined` I think

Reply:
damn you're right! typescript failed me

Reply:
Updated

Reply:
👍